flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java

```diff
@@ -107,11 +112,18 @@ public FlinkCatalog(
     this.catalogLoader = catalogLoader;
     this.baseNamespace = baseNamespace;
     this.cacheEnabled = cacheEnabled;
+    this.partitionFunctions = new LinkedHashMap<>();
```

Comment:
Nit: Does this have to be a LinkedHashMap? Normally we use the relocated Guava map constructors like `Maps.newHashMap()` or maybe the variant with expected size.